src/login/logind-session-device.c

```diff
@@ -367,9 +359,39 @@ int session_device_new(Session *s, dev_t dev, SessionDevice **out) {
         r = hashmap_put(s->devices, &sd->dev, sd);
         if (r < 0) {
                 r = -ENOMEM;
+                free(sd->node);
                 goto error;
         }
 
+        LIST_PREPEND(sd_by_device, sd->device->session_devices, sd);
+error:
+        return r;
```

Comment:
if there's only a single "return r" left in the goto block, I think it's much nicer to just return directly rather than involve goto